qiskit/optimization/algorithms/goemans_williamson_optimizer.py

```diff
@@ -0,0 +1,243 @@
+    """
+    Goemans-Williamson algorithm to approximate the max-cut of a problem.
+    The quadratic program for max-cut is given by:
+
+    max sum_{i,j<i} w[i,j]*x[i]*(1-x[j])
+
+    Therefore the quadratic term encodes the negative of the adjacency matrix of
+    the graph.
+    """
+
+    def __init__(self, num_cuts: int, sort_cuts: bool = True,
+                 unique_cuts: bool = True, seed: int = 0):
+        """
+        Args:
+            num_cuts: Number of cuts to generate.
+            sort_cuts:
+            unique_cuts: The solve method returns only unique cuts, thus there may be less cuts
+                than ``num_cuts``.
+            seed: A seed value for the random number generator.
+        """
+        super().__init__()
+
+        self._num_cuts = num_cuts
+        self._sort_cuts = sort_cuts
+        self._unique_cuts = unique_cuts
+        np.random.seed(seed)
+
+    # todo: implement
+    def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
+        """Checks whether a given problem can be solved with the optimizer implementing this method.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            Returns the incompatibility message. If the message is empty no issues were found.
+        """
+        raise NotImplemented
+
+    def solve(self, problem: QuadraticProgram) -> OptimizationResult:
+        """
+        Returns a list of cuts generated according to the Goemans-Williamson algorithm.
+
+        Args:
+            problem: The quadratic problem that encodes the max-cut problem.
+
+        Returns:
+            cuts: A list of generated cuts.
+        """
+        adj_matrix = self._extract_adjacency_matrix(problem)
+
+        chi = self._solve_max_cut_sdp(adj_matrix)
+
+        cuts = self._generate_random_cuts(chi, len(adj_matrix))
+
+        solutions = [(cuts[i, :], cut_value(cuts[i, :], adj_matrix)) for i in range(self._num_cuts)]
+
+        if self._sort_cuts:
+            solutions.sort(key=lambda x: -x[1])
+
+        if self._unique_cuts:
+            solutions = self._get_unique_cuts(solutions)
+
+        solutions = solutions[:self._num_cuts]
+        return GoemansWilliamsonOptimizationResult(x=solutions[0][0],
+                                                   fval=solutions[0][1],
+                                                   variables=problem.variables,
+                                                   status=OptimizationResultStatus.SUCCESS,
+                                                   all_solutions=solutions,
+                                                   sdp_solution=chi)
+
+    def _get_unique_cuts(self, solutions: List[Tuple[np.ndarray, float]]) \
+            -> List[Tuple[np.ndarray, float]]:
+        """
+        Returns:
+            Unique GW cuts.
+        """
+
+        # Remove symmetry in the cuts to chose the unique ones.
+        # todo: why do we need this?
```

Comment:
if you use these cuts in WS-RQAOA you may only want to run the QAOA for the unique cuts. The user can determine whether or not he/she wants this with the `unique_cuts` passed to `init`.

Reply:
Ok, clear now.